core/shared/src/main/scala/zio/logging/LogPattern.scala

```diff
@@ -0,0 +1,368 @@
+/*
+ * Copyright 2019-2023 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package zio.logging
+
+import zio.parser.{ Syntax, _ }
+import zio.{ Chunk, Config }
+
+import java.time.format.DateTimeFormatter
+import scala.util.Try
+
+/**
+ * A `LogPattern` is string representation of `LogFormat`
+ */
+sealed trait LogPattern {
```

Comment:
I wonder if we should call this `SerializableLogFormat`, or perhaps, `LogFormat.Serializable` or something.

The presence of `toLogFormat` suggests that it might be or should be a subtype of `LogFormat`.

Expressing that relationship in the name could be useful. Also moving it to the companion object of `LogFormat` could be useful if it's not as commonly used as `LogFormat`, and we want to hide it a bit or emphasize it's a derivative of `LogFormat`.